internal/lint/checker_build.go

```diff
@@ -0,0 +1,25 @@
+package lint
+
+import (
+	"context"
+)
+
+func checkerBuild(ctx context.Context, _ string) *checkResult {
```

Comment:
I find this name confusing. It is called `checkerBuild` but it seems to be checking only the module path.

Reply:
The concept is:
- the controller uses `state` optionally, not required
- the controller never modifies `state`
- the state "getter" functions either return a cached value or generate the value on the fly, cache it, and return it

The consequence of this is that the "build" checker "only" gets the path of the built exe.

Reply:
I think the concept is clear, but I still find it confusing that the build check is not building but asking for th exec path (that implicitly builds the binary). I wonder if, for this particular case, the build artifact has to be shared in the context?